src/kbs2.jl

```diff
@@ -53,47 +53,42 @@ function forceconfluence!(rs::RewritingSystem, stack, i::Integer, j::Integer, o:
 end
 
 """
-    knuthbendix2(rs::RewritingSystem, o::Ordering, maxrules::Integer)
-Implements a Knuth-Bendix algorithm that yields reduced, confluent rewriting
-system. See [Sims, p.77].
+    knuthbendix2!(rws::RewritingSystem, o::Ordering, maxrules::Integer)
+Implements the Knuth-Bendix completion algorithm that yields a reduced,
+confluent rewriting system. See [Sims, p.77].
 
 Warning: forced termiantion takes place after the number of rules stored within
 the RewritngSystem reaches `maxrules`.
 """
-function knuthbendix2!(rs::RewritingSystem, o::Ordering = ordering(rs); maxrules::Integer = 100)
-    stack = empty(rs)
-    tmprs = empty(rs)
-    for (lhs, rhs) in rules(rs)
-        push!(stack, lhs => rhs)
-        deriverule!(tmprs, stack)
-    end
+function knuthbendix2!(rws::RewritingSystem, o::Ordering = ordering(rws); maxrules::Integer = 100)
+    stack = copy(rules(rws)[active(rws)])
```

Comment:
I believe this is wrong in general. In the former the use of `deriverule!(tmprs, stack)` was done after pushing each rule from original RewritingSystem to the stack. I am almost sure that this makes a huge difference, but will have to think a bit for the particular test showing that this is wrong.

Reply:
Indeed I changed this, since I believe the former was incorrect, or rather non-optimal; if you investigate the old behavior it turns out that `deriverule!` is only called with `length(stack) == 1`!

Reply:
I looked again into it. You are right - everything works fine in that case.